src/org/labkey/test/tests/ProjectSettingsTest.java

```diff
@@ -20,61 +20,175 @@
+            goToHome();
+        }
 
         BootstrapMenu menu = new SiteNavBar(getDriver()).userMenu();
         menu.expand();    // the support and help links are now on the user menu
         List<WebElement> visibleLinks = menu.findVisibleMenuItems();
-        assertEquals("Support link state unexpected.", supportLinkPresent, visibleLinks.stream().anyMatch((a)-> a.getText().equals("Support")));
-        assertEquals("Help link state unexpected.", helpLinkPresent, visibleLinks.stream().anyMatch((a)-> a.getText().equals("LabKey Documentation")));
+        checker().verifyEquals("Support link state unexpected.",
+                supportLinkPresent, visibleLinks.stream().anyMatch((a)-> a.getText().equals("Support")));
+        checker().verifyEquals("Help link state unexpected.",
+                helpLinkPresent, visibleLinks.stream().anyMatch((a)-> a.getText().equals("LabKey Documentation")));
+
+        // If there were no errors no screenshot will be taken, returning the "not" of the screenshot will return
+        // true if there were no errors.
+        return !checker().screenShotIfNewError("Menu_Error");
+    }
+
+    private boolean checkDataInList(String projectName, String listName, List<Map<String, String>> expectedValues)
+    {
+        goToProjectHome(projectName);
+
+        Locator listLink = Locator.tagWithId("table", "lists").descendant("a").withText(listName);
+
+        waitForElement(listLink, 10_000, true);
+        clickAndWait(listLink);
+
+        DataRegionTable actualValues = new DataRegionTable("query", getDriver());
+
+        int rowIndex = 0;
+        for(Map<String, String> listRow : expectedValues)
+        {
+            checker().verifyEquals(String.format("Row %d in list '%s' is not as expected.", rowIndex, listName),
+                    listRow, actualValues.getRowDataAsMap(rowIndex));
+            rowIndex++;
+        }
+
+        // checker().screenShotIfNewError returns true if it took a screenshot, which means there is an error.
+        return !checker().screenShotIfNewError("List_Data_Error");
+    }
+
+    private void checkSettingPageValues(BaseSettingsPage settingsPage, boolean helpMenu, String supportLink,
+                                        String dateFormat, String dateTimeFormat, String timeFormat)
+    {
+
+        if(helpMenu)
+        {
+            checker().verifyTrue("Help menu should be checked.",
+                    settingsPage.getHelpMenu());
+        }
+        else
+        {
+            checker().verifyFalse("Help menu should not be checked.",
+                    settingsPage.getHelpMenu());
+        }
+
+        if(supportLink.isEmpty())
+        {
+            checker().verifyTrue("Support link should be empty.",
+                    settingsPage.getSupportLink().isEmpty());
+        }
+        else
+        {
+            checker().verifyEquals("Support link value not as expected.",
+                    supportLink, settingsPage.getSupportLink());
+        }
```

Comment:
Conditionalizing all of these assertions makes this look much more complicated that they actually are. These each can be done in a single line.
```suggestion
        checker().verifyEquals("Help menu should be " + (helpMenu ? "checked" : "unchecked"), helpMenu, settingsPage.getHelpMenu());
        checker().verifyEquals("Support link value not as expected.", supportLink, settingsPage.getSupportLink());
        //etc.
```

Reply:
Changed.